File: backend/core/src/main/java/com/onair/hearit/core/domain/exception/DomainException.java
package com.onair.hearit.core.domain.exception;

import lombok.Getter;

@Getter
public abstract class DomainException extends RuntimeException {

    private final DomainErrorCode errorCode;

    protected DomainException(DomainErrorCode errorCode, String message) {
        super(message);
        this.errorCode = errorCode;
    }
}

File: android/app/src/main/java/com/onair/hearit/domain/exception/DomainException.kt
package com.onair.hearit.domain.exception

sealed class DomainException(
    message: String,
) : Exception(message) {
    data object NetworkConnection : DomainException(ERROR_CHECK_NETWORK_MESSAGE)

    data object UserNotRegistered : DomainException(ERROR_UNAUTHORIZED_MESSAGE)
}

private const val ERROR_CHECK_NETWORK_MESSAGE = "네트워크 연결을 확인해주세요"
private const val ERROR_UNAUTHORIZED_MESSAGE = "로그인이 필요합니다"
